Merging the remaining changes to prepare the package for CRAN

janaobsteter opened the pull request.

janaobsteter reviewed (commented):
The factor in the "caste" was changed to "character", since the factor thing was making it slow and used memory. Also, I created generic functions to test whether we can improve the speed

gregorgorjanc commented:
@janaobsteter I am seeing this R CMD check (via GitHub action) failing with

```
Quitting from lines 618-622 [colony_strength] (F_Quantitative_Genetics.Rmd)
[308](https://github.com/HighlanderLab/SIMplyBee/actions/runs/9970676222/job/27550092259?pr=581#step:5:310)
Error: Error: processing vignette 'F_Quantitative_Genetics.Rmd' failed with diagnostics:
[309](https://github.com/HighlanderLab/SIMplyBee/actions/runs/9970676222/job/27550092259?pr=581#step:5:311)
no slot of name "queen" for this object of class "MultiColony"
[310](https://github.com/HighlanderLab/SIMplyBee/actions/runs/9970676222/job/27550092259?pr=581#step:5:312)
--- failed re-building ‘F_Quantitative_Genetics.Rmd’
[311](https://github.com/HighlanderLab/SIMplyBee/actions/runs/9970676222/job/27550092259?pr=581#step:5:313)
--- re-building ‘G_Sampling_functions.Rmd’ using rmarkdown
[312](https://github.com/HighlanderLab/SIMplyBee/actions/runs/9970676222/job/27550092259?pr=581#step:5:314)
--- finished re-building ‘G_Sampling_functions.Rmd’
[313](https://github.com/HighlanderLab/SIMplyBee/actions/runs/9970676222/job/27550092259?pr=581#step:5:315)
--- re-building ‘Z_FAQ.Rmd’ using rmarkdown
[314](https://github.com/HighlanderLab/SIMplyBee/actions/runs/9970676222/job/27550092259?pr=581#step:5:316)
--- finished re-building ‘Z_FAQ.Rmd’
[315](https://github.com/HighlanderLab/SIMplyBee/actions/runs/9970676222/job/27550092259?pr=581#step:5:317)
SUMMARY: processing the following file failed:
[316](https://github.com/HighlanderLab/SIMplyBee/actions/runs/9970676222/job/27550092259?pr=581#step:5:318)
  ‘F_Quantitative_Genetics.Rmd’
[317](https://github.com/HighlanderLab/SIMplyBee/actions/runs/9970676222/job/27550092259?pr=581#step:5:319)
Error: Error: Vignette re-building failed.
```

gregorgorjanc commented on tests/testthat/test-L2_colony_functions.R:
```diff
@@ -507,6 +507,27 @@ test_that("swarm", {
 })
 
+
+test_that("swarm1", {
```
Do we need a test for `buildUp1` too @janaobsteter ?

gregorgorjanc reviewed (commented):
@janaobsteter looked over this - only one comment above and see the checks

gregorgorjanc commented on src/isHeterozygous.o:
This file should not be added to the repo

gregorgorjanc commented on src/RcppExports.o:
This file should not be in the repo

gregorgorjanc commented on src/SIMplyBee.so:
This file should not be in the repo

gregorgorjanc commented on R/Functions_L0_auxilary.R:
```diff
@@ -12,7 +12,6 @@
 #' @seealso \code{\link{nNULLColonies}} and \code{\link{nEmptyColonies}}
 #' @return integer
 #'
-#' @examples
```
We likely need the `@examples` bit

gregorgorjanc commented on R/Functions_L0_auxilary.R:
```diff
@@ -2770,16 +2769,30 @@ isGenoHeterozygous <- function(x) {
   # Could inline isGenoHeterozygous() here, but isGenoHeterozygous is far easier
   # to test than isCsdHeterozygous()
-  ret <- isGenoHeterozygous(x = geno)
+  ret <- isGenoHeterozygous(geno)
```
I assume thins line is redundant

gregorgorjanc reviewed (changes requested):
@janaobsteter see comments above

gregorgorjanc commented on src/RcppExports.o:
@janaobsteter this object file should be removed from git repo (add to `.gitignore` - see how Chris handles these)

gregorgorjanc commented on src/SIMplyBee.so:
@janaobsteter this object file should be removed from git repo (add to `.gitignore` - see how Chris handles these)

gregorgorjanc commented on src/isHeterozygous.o:
@janaobsteter this object file should be removed from git repo (add to `.gitignore` - see how Chris handles these)

gregorgorjanc reviewed (commented):
@janaobsteter similar comments to last time;)

gregorgorjanc commented on NEWS.md:
@janaobsteter Add a note that we added isCsdHetero... to speed things a bit?

gregorgorjanc commented on R/Functions_L0_auxilary.R:
```diff
@@ -1063,7 +1063,7 @@ getQueenYearOfBirth <- function(x, simParamBee = NULL) {
     }
   } else if (isColony(x)) {
     if(packageVersion("AlphaSimR") > package_version("1.5.3")){
```
@janaobsteter Best to just fix package version in DESCRIPTION and then we don't need this code duplication.

gregorgorjanc commented on src/RcppExports.o:
@janaobsteter remove this file from the repo - this is a compiled file

gregorgorjanc commented on src/SIMplyBee.so:
@janaobsteter remove this file from the repo - this is a compiled file

gregorgorjanc reviewed (changes requested):
@janaobsteter looks good - needs minor tweaks and then we can merge

janaobsteter commented:
@gregorgorjanc , this is now the version accepted to CRAN. I've added all anchors to \links and removed the checks for AlphaSimR version since it now depends on the latest version with the correct handling of the misc slot.

Could you please merge this into HLab, then I can merge dev into main and I will also mark this as commit 0.4.0

gregorgorjanc reviewed (approved):
LGTM @janaobsteter! Thanks for doing all this edits for the help page linking! That must have been annoying.
